trickler/mobile/lib/widgets/weight_input.dart

```diff
@@ -94,8 +94,11 @@ class WeightInputState extends State<WeightInput> {
           signed: false
         ),
         controller: controller,
+        /// TODO(cleanup): Is this necessary?
         focusNode: _getFocusNodeWithListener(),
+        /// TODO(ux): This causes issues because it sends partial values. Use onSubmit instead.
         onChanged: _setWeightFromText,
+        /// TODO(ux): Use onSubmitted instead?
```

Comment:
The logic behind using these partial values is that it would allow us to update the value with multiple inputs prior to submit, so like they could type in 1.1, and use the dial, or + - buttons to get to 1.18. Not entirely necessary, but if we ever planned on adding more inputs back in then it probably wouldn't be worth out time to refactor this logic.

Reply:
Ok. The issue right now is that it sends as the user types. So, if I'm in an active session auto-mode is on, it will change the target value in the middle of it. The use-case might be that the user is getting ready to change the setting for the next charge, but without disturbing the current one.

Reply:
Okay, I see what you are saying, a possible solution would be for the app to pause the weight writes to the peripheral `onChanged`, and then continue them `onSubmitted`. This would allow the app to stay up to date with what the user is inputing, and wouldn't affect the current measurement being run on the peripheral.